### ISSUE TITLE
Sarama: disconnected producer stays disconnected

### ISSUE THREAD

**Issue by josselin-c** [Shopify/sarama#741]
##### Versions
Sarama Version: 1.9
Kafka Version: 0.9
Go Version: 1.6.2

##### Configuration
Default sarama config

##### Logs
[Sarama] 2016/09/01 18:01:51 write tcp 55.66.77.88:56398->11.22.33.44:9092: write: broken pipe
...
[Sarama] 2016/09/02 14:26:46 write tcp 55.66.77.88:56398->11.22.33.44:9092: write: broken pipe
[Sarama] 2016/09/02 14:26:47 write tcp 55.66.77.88:56398->11.22.33.44:9092: write: broken pipe

##### Problem Description
I'm using a SyncProducer, after some connectivity issue the producer can't produce to kafka anymore and always return an error, even after the connectivity to kafka is restored.
Is this a scenario my code should handle: close the current consumer, create a new one? Or should this be handled by the producer?

Thanks

**Comment by eapache** [comment]
No, this should be handled by the producer. I took a quick look but don't see an obvious cause for this bug. Could you please provide more logs (in particular all the ones that start with `producer/leader/` and `producer/broker/`) so I can try to fix this?

**Comment by josselin-c** [comment]
Okay, after doing some tests, I can attest that production seem to restart and indeed is resilient to disconnection / network errors.

I tried to find were the error I got was coming from, I think it comes from the sarama-cluster library which uses the Sarama Logger to log its own errors (and doesn't add a "cluster" prefix or something) which confused me as I though the error was in Sarama.
I don't know why my code stopped producing but it seem unlikely to be because of the sarama producer.
